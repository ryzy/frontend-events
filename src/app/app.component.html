<div class="mat-elevation-z2">
  <button color="primary" mat-button>Click me!</button>
</div>
