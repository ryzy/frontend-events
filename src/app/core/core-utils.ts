/**
 * Prevent re-import of core modules
 *
 * @url https://angular.io/guide/styleguide#style-04-12
 *
 * Usage:
 * ```
 * export class MyModule {
 *   constructor(@Optional() @SkipSelf() parentModule: MyModule) {
 *     throwIfAlreadyLoaded(parentModule, 'MyModule');
 *   }
 * }
 * ```
 */
export function throwIfAlreadyLoaded(parentModule: any, moduleName: string): void {
  if (parentModule) {
    throw new Error(`${moduleName} has already been loaded. Import Core modules in the AppModule only.`);
  }
}

/**
 * Returns true if code is executed in test (i.e. Karma) context.
 */
export function isUnitTestContext(): boolean {
  return !!(window as any).__karma__;
}

// tslint:disable:no-bitwise
/**
 * Generate UUID string
 */
export function uuid(length?: number): string {
  const str = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c: string) => {
    const r: number = (Math.random() * 16) | 0;
    const v: number = c === 'x' ? r : (r & 0x3) | 0x8;
    return v.toString(16);
  });

  return length ? str.substr(0, length) : str;
}
// tslint:enable:no-bitwise

/**
 * Get random number from given range
 */
export function randomRange(min: number = 0, max: number = 999): number {
  return Math.floor(Math.random() * (max - min)) + min;
}
