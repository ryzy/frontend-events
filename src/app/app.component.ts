import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { User } from './core/model/user';

import { AuthService } from './core/services/auth.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent {
  public user$: Observable<User|undefined>;

  public constructor(authService: AuthService) {
    this.user$ = authService.getCurrentUser();
  }
}
