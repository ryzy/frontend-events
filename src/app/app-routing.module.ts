import { NgModule } from '@angular/core';
import { Routes, RouterModule, PreloadAllModules } from '@angular/router';

import { IsInitDataFetchedGuard } from './core/guards/is-init-data-fetched.guard';
import { NotFoundPageComponent } from './shared/containers/not-found-page/not-found-page.component';

// Application root routes
const appRoutes: Routes = [
  {
    path: '',
    loadChildren: './events-list/events-list.module#EventsListModule',
    canActivate: [IsInitDataFetchedGuard],
  },
  {
    path: 'ev',
    loadChildren: './event-details/event-details.module#EventDetailsModule',
    canActivate: [IsInitDataFetchedGuard],
  },
  {
    path: 'edit',
    loadChildren: './event-form/event-form.module#EventFormModule',
    canActivate: [IsInitDataFetchedGuard],
  },
  {
    path: 'user',
    loadChildren: './user/user.module#UserModule',
    canActivate: [IsInitDataFetchedGuard],
  },
  {
    path: '**',
    component: NotFoundPageComponent,
  },
];

@NgModule({
  imports: [
    RouterModule.forRoot(appRoutes, {
      // enableTracing: true,
      initialNavigation: 'enabled',
      anchorScrolling: 'enabled',
      scrollPositionRestoration: 'top',
      relativeLinkResolution: 'corrected',
      preloadingStrategy: PreloadAllModules,
    }),
  ],
})
export class AppRoutingModule {}
