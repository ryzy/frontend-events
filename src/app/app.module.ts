import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { MaterialModule } from './material.module';
import { ServiceWorkerModule } from '@angular/service-worker';

import { AppComponent } from './app.component';

import { environment } from '../environments/environment';
import { AngularFireModule } from 'angularfire2';
import { AngularFirestoreModule } from 'angularfire2/firestore';
import { AngularFireAuthModule } from 'angularfire2/auth';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    MaterialModule,
    AngularFireModule.initializeApp(environment.firebase),
    AngularFireAuthModule,
    AngularFirestoreModule,
    ServiceWorkerModule.register('/ngsw-worker.js', { enabled: environment.production }),
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
