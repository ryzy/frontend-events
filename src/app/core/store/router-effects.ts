import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { Location } from '@angular/common';
import { Effect, Actions, ofType } from '@ngrx/effects';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';
import { tap } from 'rxjs/operators';

import * as fromRouter from './router-actions';
import { RouterActionType } from './router-actions';

@Injectable()
export class RouterEffects {
  @Effect({ dispatch: false })
  public navigate$: Observable<any> = this.actions$.pipe(
    ofType<fromRouter.GoAction>(RouterActionType.GO),
    tap((action: fromRouter.GoAction) => {
      this.router.navigate(action.path, action.extras);
    }),
  );

  @Effect({ dispatch: false })
  public navigateBack$: Observable<Action> = this.actions$
    .pipe(
      ofType(RouterActionType.BACK),
      tap(() => this.location.back()),
    );

  @Effect({ dispatch: false })
  public navigateForward$: Observable<Action> = this.actions$
    .pipe(
      ofType(RouterActionType.FORWARD),
      tap(() => this.location.forward()),
    );

  public constructor(
    private actions$: Actions,
    private router: Router,
    private location: Location,
  ) {}
}
