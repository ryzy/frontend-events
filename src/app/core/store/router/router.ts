import { Params } from '@angular/router';
import { createFeatureSelector, createSelector, MemoizedSelector } from '@ngrx/store';
import { RouterReducerState } from '@ngrx/router-store';

/**
 * Possible URL params used in the app
 */
export interface AppRouterParams extends Params {
  eventId?: string;

  apiKey?: string; // on /user pages

  // events filters
  where?: string;
  tags?: string;
  ws?: string; // workshops
  fws?: string; // free workshops

  // sort info
  s?: string; // sort by info
  sd?: 'desc'; // sort direction. only `desc` since `asc` is by default, so it shouldn't appear in url
}

/**
 * App custom router state
 */
export interface AppRouterState {
  /**
   * Clean URL, w/o any params, query params or fragment
   */
  url: string;

  /**
   * Full URL, with whatever is present there (custom params, query params, fragment etc)
   */
  urlFull: string;

  /**
   * Any `data` added to Route config
   */
  data: { [name: string]: any };

  /**
   * Current #fragment (or empty string)
   */
  fragment: string;

  /**
   * URL :params
   */
  params: AppRouterParams;

  /**
   * URL ?query=params
   */
  queryParams: Params;
}

export const defaultAppRouterState: AppRouterState = {
  url: '/',
  urlFull: '/',
  data: {},
  fragment: '',
  params: {},
  queryParams: {},
};

export const getAppRouterState: MemoizedSelector<object, AppRouterState> = createSelector(
  createFeatureSelector<RouterReducerState<AppRouterState>>('router'),
  (state: RouterReducerState<AppRouterState>) => (state && state.state) || defaultAppRouterState,
);
