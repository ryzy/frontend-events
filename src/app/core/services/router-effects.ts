import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { Location } from '@angular/common';
import { Effect, Actions, ofType } from '@ngrx/effects';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs';
import { tap } from 'rxjs/operators';

import * as fromRouter from '../store/router/router-actions';
import { RouterActionType } from '../store/router/router-actions';

@Injectable({
  providedIn: 'root',
})
export class RouterEffects {
  @Effect({ dispatch: false })
  public navigate$: Observable<any> = this.actions$.pipe(
    ofType<fromRouter.GoAction>(RouterActionType.GO),
    tap((action: fromRouter.GoAction) => {
      this.router.navigate(action.path, action.extras);
    }),
  );

  @Effect({ dispatch: false })
  public navigateBack$: Observable<Action> = this.actions$.pipe(
    ofType(RouterActionType.BACK),
    tap(() => this.location.back()),
  );

  @Effect({ dispatch: false })
  public navigateForward$: Observable<Action> = this.actions$.pipe(
    ofType(RouterActionType.FORWARD),
    tap(() => this.location.forward()),
  );

  public constructor(private actions$: Actions, private router: Router, private location: Location) {}
}
