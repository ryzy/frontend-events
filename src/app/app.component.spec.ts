import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { Router } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';

import { AppTestingAuthAndDbModule } from '../testing/app-testing-auth-db.module';
import { AppComponent } from './app.component';
import { RouterEffects } from './core/services/router-effects';
import { SharedModule } from './shared/shared.module';

describe('AppComponent', () => {
  let app: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let router: Router;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule, AppTestingAuthAndDbModule, EffectsModule.forRoot([RouterEffects])],
      declarations: [AppComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    router = TestBed.get(Router);
    fixture = TestBed.createComponent(AppComponent);
    app = fixture.debugElement.componentInstance;
    fixture.detectChanges();
  });

  it('should create the app', async(() => {
    expect(app).toBeTruthy();
  }));

  it('should go to new event page', async(() => {
    spyOn(router, 'navigate');
    app.goToNewEvent();
    expect(router.navigate).toHaveBeenCalled();
  }));
});
