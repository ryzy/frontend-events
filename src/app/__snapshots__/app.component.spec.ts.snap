// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
<app-root
  user$={[Function Store]}
>
  <header
    class="app-header"
  >
    <div
      class="grid-container"
    >
      <div
        class="grid-x align-middle align-justify"
      >
        <div
          class="cell mini-8"
        >
          <h1
            class="app-header__logo cell"
          >
            <a
              href="/"
              ng-reflect-router-link="/"
              routerlink="/"
            >
              What Conference Next
            </a>
          </h1>
        </div>
        <div
          class="cell mini-4 text-right app-menu"
        >
          <a
            class="app-menu-item"
            href="/edit"
            ng-reflect-router-link="/edit"
            routerlink="/edit"
          >
            New Event
          </a>
          <a
            aria-disabled="false"
            class="app-menu-item app-menu-item--user mat-icon-button mat-raised-button mat-button-base _mat-animation-noopable"
            href="/user"
            mat-icon-button=""
            mat-raised-button=""
            ng-reflect-router-link="/user"
            routerlink="/user"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <app-user-avatar
                class="app-user-avatar"
              >
                
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-icon-no-color"
                  role="img"
                >
                  perm_identity
                </mat-icon>
                
              </app-user-avatar>
            </span>
            <div
              class="mat-button-ripple mat-ripple mat-button-ripple-round"
              matripple=""
              ng-reflect-centered="true"
              ng-reflect-disabled="false"
              ng-reflect-trigger="https://github.com/user"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
        </div>
      </div>
    </div>
  </header><div
    class="grid-container app-main-container"
  >
    <router-outlet />
  </div><footer
    class="app-footer"
  >
    <div
      class="grid-container"
    >
      <div
        class="grid-x small"
      >
        <div
          class="cell medium-7"
        >
           © What Conference Next - easily find the next technical event for you 
        </div>
        <div
          class="cell medium-5 text-right"
        >
           Made with 
          <span
            class="color-accent"
          >
            ❤
          </span>
           by 
          <a
            href="https://twitter.com/ryzmen"
            target="_blank"
          >
            @ryzmen
          </a>
             |   
          <a
            href="mailto:hi@what-conference-next.com"
          >
            Contact
          </a>
        </div>
      </div>
    </div>
  </footer>
</app-root>
`;
